src/services/classifier2020.ts

```diff
@@ -225,6 +225,10 @@ namespace ts.classifier.v2020 {
         return (isQualifiedName(node.parent) && node.parent.right === node) || (isPropertyAccessExpression(node.parent) && node.parent.name === node);
     }
 
+    function isInfinityOrNaNString(name: __String): boolean {
+        return name === "Infinity" || name === "-Infinity" || name === "NaN";
+    }
+

```

Comment:
Not sure if there's a better way to check for `Infinity` and `NaN`, I copied this from `checker.ts` line 26202.
Let me know if there's a better way to fix this.